frontend/src/pages/ActivityReport/Pages/nextSteps.js

```diff
@@ -1,24 +1,200 @@
+      required={isRequired}
+      name={name}
+      label="What have you agreed to do next?"
+    >
+      <TextInput name={name} onChange={onUpdate} data-testid={`${name}-input`} defaultValue={input} />
+      <Button outline disabled={!(input && input.trim())} onClick={onSubmit} data-testid={`${name}-button`} type="button">Save Next Step</Button>
+      {!isRequired && <Button secondary onClick={onCancel} type="button" data-testid={`${name}-cancel-button`}>Cancel</Button>}
+    </FormItem>
+  );
+};
+
+NoteEntry.propTypes = {
+  onEntry: PropTypes.func.isRequired,
+  onCancel: PropTypes.func.isRequired,
+  name: PropTypes.string.isRequired,
+  defaultValue: PropTypes.string,
+  isRequired: PropTypes.bool,
+};
+
+NoteEntry.defaultProps = {
+  isRequired: false,
+  defaultValue: '',
+};
+
+const NoteEntries = ({ name, humanName }) => {
+  const {
+    register, control, setValue, trigger,
+  } = useFormContext();
+  const notes = useWatch({ name, control });
+
+  const [showPrompt, updateShowPrompt] = useState(false);
+  const [targetIndex, updateTargetIndex] = useState(-1);
+
+  useEffect(() => {
+    register({ name }, { validate: (allNotes) => (allNotes.length !== 0 ? true : `${humanName} requires at least one step`) });
+  });
+
+  const onEntry = (note, index, noteId) => {
+    const newNotes = [...notes];
+    newNotes[index] = { note, id: noteId };
+    setValue(name, newNotes);
+    updateShowPrompt(false);
+    trigger(name);
+  };
+
+  const onEdit = (index) => {
+    updateShowPrompt(true);
+    updateTargetIndex(index);
+    trigger(name);
+  };
+
+  const onDelete = (index) => {
+    const newNotes = notes.filter((item, itemIndex) => itemIndex !== index);
+    setValue(name, newNotes);
+    updateShowPrompt(false);
+    trigger(name);
+  };
+
+  const onCancel = () => {
+    updateShowPrompt(false);
+  };
+
+  if (notes.length === 0) {
+    return (
+      <Fieldset className="smart-hub--report-legend smart-hub--form-section" legend={`${humanName} Next Steps`}>
+        <NoteEntry
+          onEntry={(value) => onEntry(value, 0)}
+          isRequired
+          name={name}
+          onCancel={onCancel}
+          humanName={humanName}
+        />
+      </Fieldset>
+    );
+  }
+
+  const targetId = notes[targetIndex] ? notes[targetIndex].id : undefined;
+  const defaultValue = notes[targetIndex] ? notes[targetIndex].note : undefined;
+
+  return (
+    <>
+      <Fieldset className="smart-hub--report-legend smart-hub--form-section" legend={`${humanName} Next Steps`}>
+        <ul className="usa-list--unstyled">
+          {notes.map((item, index) => (
+            <li key={item.note} className="grid-row flex-row border-bottom padding-top-2 padding-bottom-2" style={{ borderColor: '#f0f0f0' }}>
+              <div className="grid-col flex-12">
+                {item.note}
+              </div>
+
+              <div className="grid-col" style={{ marginTop: '0px' }}>
+                <ContextMenu
+                  label="Menu"
```

Comment:
`Menu` is a bit too generic to be useful to screenreaders. Maybe "edit menu" or "actions menu"?